prusti-tests/tests/verify/pass/terminating_fib.rs

```diff
@@ -0,0 +1,16 @@
+// compile-flags: -Punsafe_core_proof=true
+
+#![allow(unused)]
+
+use prusti_contracts::*;
+
+fn main() {}
+
+#[pure(Int::new(x))]
```

Comment:
What is this? Is this the termination measure? Why is it attached to the `pure` attribute?

Reply:
yes, it is a termination measure.
The PR makes it a requirement for recursive pure functions, or recursive functions with the `#[terminates]` attribute.
I also specifically added it to the `#[pure]` attribute because nonterminating pure expressions in pre/post conditions can cause unsoundness

Reply:
Still I don't think the UX is good. It should be clear when reading the contract of a method at least roughly what it means. So, if a termination measure is required for a pure function and there is none provided, we should tell the user to write a `#[terminates(...)]` attribute.